STACpopulator/populator_base.py

```diff
@@ -55,6 +68,28 @@ def __init__(
         LOGGER.info(f"Collection {self.collection_name} is assigned ID {self.collection_id}")
         self.create_stac_collection()
 
+    @staticmethod
+    def _load_extra_parser(func_str: str, extra_kwargs: dict[str, str]) -> Callable:
+        if ":" in func_str:
+            mod, func = func_str.split(":", 1)
+            try:
+                function_ns = importlib.import_module(mod)
```

Comment:
I think this will fail unless it is a direct file (same PWD module) or that it is placed in the sys path before (as shown in the tests with `sys.path.append(str(request.path.parent / "extra_functions"))`. 

Since the original intent is to add Magpie/Cowbird extra metadata, I expect those functions would be placed is some Python files at various locations either in a complex dir hierarchy or in installed nested-module. It is better if this resolution can be done automatically without need from the user to patch the paths. 

This is how I've dealt with these combinations in Weaver: https://github.com/crim-ca/weaver/blob/8bf64a83ca894ea88dffd5e90e2b63ec3d5b268c/weaver/utils.py#L1060